### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by azmeuk**
This pull requests does various things:
- Fixed tests with `env TZ=UTC+5 python2 setup.py test`
- Add travis support (tests python 2.7 and everything from python 3.4 to 3.8-dev)
- A bit of python 3 compatibility (avoid repr backtick syntax, new print function etc...)

So the point is to check that python 2 tests are still valid, as a base to make them work with python 3.

### CODE REVIEW

File: .travis.yml
Comment (mgedmin) [inline .travis.yml]: Python 3.4 is EOL and we're (slowly) dropping support for it everywhere.

File: setup.cfg
Comment (mgedmin) [inline setup.cfg]: Please drop the `{` and `}`

File: setup.cfg
Comment (mgedmin) [inline setup.cfg]: Lack of spaces around `=` looks weird here, considering they're everywhere else.

File: setup.cfg
Comment (mgedmin) [inline setup.cfg]: I would drop the spaces around the second `=` to get `setenv = TZ=UTC+5`.

(Also it's a bit sad if the tests only pass in one specific timezone, but I don't think that's your fault?)

File: src/zc/FileStorage/snapshotintime.py
Comment (mgedmin) [inline src/zc/FileStorage/snapshotintime.py]: Uh, please don't do this?

There's proper API to convert a TimeStamp to an 8-bytestring.  IIRC it's `.raw()`?  At least in the modern `persistent.timestamp.TimeStamp` implementation.  I think ZODB.TimeStamp is an alias to that one nowadays.

File: src/zc/FileStorage/tests.py
Comment (mgedmin) [inline src/zc/FileStorage/tests.py]: You can drop the trailing comma, or replace it with a `.rstrip('\n')`.

Comment (mgedmin) [reply src/zc/FileStorage/tests.py]: Also, this looks like you're trying to print a tuple?  Which is not what the test expects on Python 2, and not how the print function acts on Python 3?  So maybe drop the space in front of the `(`?

File: src/zc/FileStorage/tests.py
Comment (mgedmin) [inline src/zc/FileStorage/tests.py]: Same here.

File: src/zc/FileStorage/tests.py
Comment (mgedmin) [inline src/zc/FileStorage/tests.py]: Same here.

File: src/zc/FileStorage/tests.py
Comment (mgedmin) [inline src/zc/FileStorage/tests.py]: There are a lot of these `print` space `(` things that should not have spaces.

File: .travis.yml
Comment (mgedmin) [inline .travis.yml]: It doesn't look like Travis CI honors these.

The documented syntax is [quite different](https://docs.travis-ci.com/user/customizing-the-build#rows-that-are-allowed-to-fail).

I would also suggest using a global `dist: xenial` and listing Python versions at the top level, because apparently allow_failures [doesn't work for jobs defined with `matrix.include` or something](https://docs.travis-ci.com/user/customizing-the-build#matching-jobs-with-allow_failures)?  The docs are not very clearly written.

Note that in that case you'll have to refer to pypy/pypy3.5 using fully-qualified versions such as pypy.2.7-6.0 and pypy3.5-6.0 because Travis CI is *gesticulates wildly bereft of speech*.

**Review by mgedmin — COMMENTED** [review]
Needs some work, specifically

- let's make Travis happy
- that repr().replace(...) bit needs fixing

I'm not setting the status to "request changes" because I hate it when people do that and then you make changes and they don't come back to update their reviews, and I cannot guarantee I'll personally have the energy to re-review this again.  (Someone else ought to, or I might.)

File: setup.cfg
Comment (azmeuk) [reply setup.cfg]: Without this timezone, `zc.FileStorage.tests.test_pack_sleep` does not pass. It is too bad indeed, but in a first time, I just tried to make the tests pass without changing anything.

**Comment by azmeuk** [comment]
Thank you for your review @mgedmin . All the things you pointed are now fixed :)

File: .travis.yml
Comment (mgedmin) [inline .travis.yml]: Please drop this: it's no longer supported by Travis CI.

File: setup.py
Comment (mgedmin) [inline setup.py]: I would prefer if we kept the `[test]` extra: there's overall movement in zopefoundation/* to ditch `python setup.py test` and start using zope-testrunner directly, for which installing the tests with `pip install -e .[test]` or via tox's `extras = test` is very helpful.

**Review by mgedmin — APPROVED** [review]

File: setup.py
Comment (azmeuk) [reply setup.py]: I would like to use `zope-testrunner` in another pull request. What do you think?

File: .travis.yml
Comment (azmeuk) [reply .travis.yml]: Fixed.

File: setup.py
Comment (jamadden) [reply setup.py]: I think it's fine to defer changing the testrunner to a later PR. But the `test` extra needs to stay (be returned) in this PR.

Comment (mgedmin) [reply setup.py]: Sure, that was a description of the general trends, not a requirement for this PR.

Comment (azmeuk) [reply setup.py]: I restored `test` extra, but I also kept `test_require` for now so `python setup.py test` still work.

**Review by mgedmin — APPROVED** [review]
LGTM!

**Comment by mgedmin** [comment]
So, the usual workflow is people merge their own pull requests once they get their PRs approved by somebody else.

Anyone can get commit permissions if they sign the the [Zope Foundation Committer Agreement](https://docs.zope.org/developer/becoming-a-committer.html)?  (Aside: as a signatory myself, I'm not allowed to merge code from people who haven't signed it.  There are exceptions for trivial changes that aren't considered copyrightable, but I don't feel myself qualified to make the distinction so I let somebody else merge those.)

Now the [link](https://docs.zope.org/developer/becoming-a-committer.html) I had that describes the process is, uh, broken???  Looks like the TLS certificate for docs.zope.org expired and nobody noticed.  Who do I ping about that?  (I'm always bad about remembering who to contact, or which repository to file issues in.)

Also, I've seen in other zopefoundation/ PRs that people have emailed scanned signed committer agreement PDF printouts and never received an answer, so I don't know what to tell you.  Zope Foundation is being dissolved and all the copyrights are getting transferred to the Plone Foundation, but AFAIU that hasn't happened yet so the old processes should be working still.

Can anyone help?

**Comment by azmeuk** [comment]
> Also, I've seen in other zopefoundation/ PRs that people have emailed scanned signed committer agreement PDF printouts and never received an answer, so I don't know what to tell you. Zope Foundation is being dissolved and all the copyrights are getting transferred to the Plone Foundation, but AFAIU that hasn't happened yet so the old processes should be working still.

I just sent the signed agreement anyway :)

**Comment by azmeuk** [comment]
@mgedmin I am now a member of the Zope group. However, github does not show me any button to merge this pull request. Can you do it?

**Comment by mgedmin** [comment]
Let's get this figured out properly: I see you're a member of the (new to me) Contributors team, which has access to 0 repositories (which seems wrong!).  The old Developers team has access to 356 repositories.  (We have 384 repositories total, so now I wonder what are the ones Developers didn't have access to?)

AFAIU this is happening because of the transition to the Plone Foundation and the new contributor agreement.

@polyester, how are things supposed to work nowadays?  Should Contributors have write access to all (or at least most) of the repositories?

**Comment by polyester** [comment]
@mgedmin ah, I had assumed the access had been set up. Yes, the Contributors team should have write access, that's where everybody goes in that has re-signed the Contributor Agreement

**Comment by mgedmin** [comment]
@polyester Do you know who created that group?  Or how to set up repository access?  I'm poking around the GitHub UI, but all I can find is a form to add the repositories one by one, and I don't think that's very scalable.

(Also, I don't want to step on anybody's toes by fiddling with random knobs!)

**Comment by polyester** [comment]
@mgedmin the group was created by @MatthewWilkes, I'm now filling it up as I'm receiving the CLA's. I'll poke around to see if there's an easy script to give write access; it's easy enough to get a list of all repos via the api but I'll have to see how to *set* things with the api. I'm sure someone has done that already, will check later.

**Comment by mgedmin** [comment]
It looks like [zope.githubsupport](https://github.com/zopefoundation/zope.githubsupport/blob/master/src/zope/githubsupport/repos.py#L154-L168) has code to update teams for a repository.  It's woefully underdocumented, though.

**Comment by MatthewWilkes** [comment]
@polyester @mgedmin So, we've got lots of historical stuff with the zopefoundation org, as it predated a lot of the better support in GitHub for teams. I suspect the zope.githubsupport repo is a bit out of date in terms of what's now expected in GitHub.

It looks like in the plone org all the repos are added to the developer team, I expected this would be done by mr.roboto, but it doesn't seem to be. Would someone on the AI team know?

**Comment by polyester** [comment]
Maybe. Let's ask around. Also, there's a number of repos in the zopefoundation that don't allow the 'developers' team to push. Could be because they are old, could be for another reason. I'm new around these parts, don't want to step on people's toes...
It is probably easier to do a script to add all repos to the new 'contributors' team, but maybe there's a reason these were excluded. 
The ones that don't give 'developers' access are:

"https://github.com/zopefoundation/developer_docs"
"https://github.com/zopefoundation/js.codemirror"
"https://github.com/zopefoundation/Products.FSDump"
"https://github.com/zopefoundation/www.zope.org"
"https://github.com/zopefoundation/z3c.authviewlet"
"https://github.com/zopefoundation/z3c.i18n"
"https://github.com/zopefoundation/z3c.locales"
"https://github.com/zopefoundation/z3c.preference"
"https://github.com/zopefoundation/z3c.rmldocument"
"https://github.com/zopefoundation/z3c.testsummarizer"
"https://github.com/zopefoundation/z3c.wizard"
"https://github.com/zopefoundation/zc.authorizedotnet"
"https://github.com/zopefoundation/zc.datetimewidget"
"https://github.com/zopefoundation/zc.FileStorage"
"https://github.com/zopefoundation/zc.intid"
"https://github.com/zopefoundation/zc.recipe.cmmi"
"https://github.com/zopefoundation/zc.set"
"https://github.com/zopefoundation/zmi.styles"
"https://github.com/zopefoundation/zope2docs"
"https://github.com/zopefoundation/zope2.sessioncookie"
"https://github.com/zopefoundation/zope.app.catalog"
"https://github.com/zopefoundation/zope.app.file"
"https://github.com/zopefoundation/zope.app.onlinehelp"
"https://github.com/zopefoundation/zope.app.preference"
"https://github.com/zopefoundation/zope.app.renderer"
"https://github.com/zopefoundation/zope.broken"
"https://github.com/zopefoundation/zopefoundation.github.io"
"https://github.com/zopefoundation/zope.xmlpickle"

**Comment by icemac** [comment]
@polyester Some of the repositories in the list are already archived (like developer_docs or zmi.styles) or could be archived (like zope2docs).
IMHO for the other ones it got forgotten to add access for the developers because until now only admins have pushed to them.

**Comment by polyester** [comment]
OK, giving team "Contributors" write access to all repos now. Script should be finished in a bit.

for future me: [this script](https://gist.github.com/netflash/381f03978620fabd8ac9449dfa6c639b) is the easy way

**Comment by azmeuk** [comment]
It worked. Thank you all for your help!

**Pull request merged**
